### PR TITLE
Improve Stripe KYC flow for Paid Newsletters

### LINKED ISSUE #72282: [Earn] Figure out how to streamline Stripe connection flow from the Subscribe block
We have a following problem after merging the newsletters branch:

Stripe Connection flow is very confusing:

1. Click Add plan from the Subscribe Block
2. Complete KYC
3. Click the back button from WPCOM
4. User ends up on the Earn page

The scope of this task is to:
a) Record a video of this problem so we can discuss it better
b) Do preliminary research of what is possible. Maybe we can somehow save the state before redirecting to KYC so we redirect again to a proper add product flow?

For the purposes of this task we can assume we are already on a paid plan, but I would love a video on both a paid and a free plan with the whole upgrade path as well.

### PR DESCRIPTION
**Pull request by romarioraffington**
**Note to self: Do not deploy until https://github.com/Automattic/jetpack/pull/29351 is released to WPCOM** 

Resolves https://github.com/Automattic/wp-calypso/issues/72282
Relates to  https://github.com/Automattic/jetpack/pull/29351 

## Proposed Changes

1.  When  query param, `?stripe_connect_success=earn-newsletter`, is available on the Earn page, display a Newsletter specific notice that prompt the user to create a newsletter payment plan after completing Stripe KYC

Screenshot 2023-03-06 at 9 54 03 PM

2. When "Add Payments" is clicked, open a new tab with the following URL: `earn/payments-plans/romarioraffearn.wordpress.com#add-newsletter-payment-plan,` which opens a modal to add a payment plan

Screenshot 2023-03-06 at 9 53 42 PM

3. Update the "Add Payment Plan" modal header to be newsletter specific if the `#add-newsletter-payment-plan` hash is available. 

| Adding a Newsletter Plan  | Editing a Newsletter Plan |
| ------------- | ------------- |
| Screenshot 2023-03-06 at 9 53 42 PM  | Screenshot 2023-03-06 at 9 53 50 PM  |

## Testing Instructions

1. Go to `http://calypso.localhost:3000/earn/payments/romarioraffearn.wordpress.com?stripe_connect_success=earn-newsletter` and verify the notice with the following message is displayed: "Congrats! Your site is now connected to Stripe. You can now add payments to your newsletter."
2. Click on the "Add payments" CTA in the Notice
3. Verify that a new tab is opened and the modal is displayed to add a Newsletter Plan
4. Verify that the title of the modal is "Add a newsletter payment plan"
5. Create a newsletter payment plan
6. Edit the newsletter payment plan and verify the title of the modal is "Edit a newsletter payment plan"

### Demo

Appears that my microphone wasn't recording but the video should still be of value.

https://www.loom.com/share/cba348e10b4d4383aab2855ef1aa4822

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/earn/memberships/products.jsx
Comment (and1can) [inline client/my-sites/earn/memberships/products.jsx:36]: Note to self: the values here are refactored into constants file.

**Comment by and1can** [comment]
Hi @romarioraffington, 

Great work on this fix! Just a nit that if you change `Related` to `Fixes` in your PR description, the parent ticket will then be automatically closed once this PR merges. 

See link: https://github.com/Automattic/wp-calypso/issues/68941#issuecomment-1451432278

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7845883

Thank you @romarioraffington for including a screenshot in the description! This is really helpful for our translators.

**Comment by romarioraffington** [comment]
> Great work on this fix! Just a nit that if you change Related to Fixes in your PR description, the parent ticket will then be automatically closed once this PR merges.

Ah yes, the parent ticket requires fixes in other repos as well ( like Jeptack and WPCOM ) . I want to keep the parent ticket open until all the fixes are in

File: client/my-sites/earn/memberships/index.jsx
Comment (millerf) [inline client/my-sites/earn/memberships/index.jsx:501]: Nice move!

File: client/my-sites/earn/memberships/index.jsx
Comment (millerf) [inline client/my-sites/earn/memberships/index.jsx]:
```suggestion
		source: window.location.hash === ADD_NEWSLETTER_PAYMENT_PLAN_HASH ? 'earn-newsletter' : 'calypso',
```

File: client/my-sites/earn/memberships/index.jsx
Comment (millerf) [inline client/my-sites/earn/memberships/index.jsx:514]: 'earn' => Shouldn't that be calypso?

I haven't tested, but that seems like it should break the original flow by not diaplaying the notice....

Can we introduce constants for that as well?

**Review by millerf — CHANGES_REQUESTED** [review]
The idea is great but from my understanding it is breaking the "original" flo by not displaying the `'Congrats! Your site is now connected to Stripe. You can now add your first payment plan.'` notice...

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7845883).

File: client/my-sites/earn/memberships/index.jsx
Comment (romarioraffington) [reply client/my-sites/earn/memberships/index.jsx:501]: > The idea is great but from my understanding it is breaking the "original" flo by not displaying the 'Congrats! Your site is now connected to Stripe. You can now add your first payment plan.' notice...

Great question, the original message will be displayed when the `'#add-newsletter-payment-plan`  hash is not in the URL as we are determining the source to send to the backend based on that hash.

File: client/my-sites/earn/memberships/index.jsx
Comment (romarioraffington) [reply client/my-sites/earn/memberships/index.jsx]: Fixed in [adce253](https://github.com/Automattic/wp-calypso/pull/74146/commits/adce253dadc976f5803a918ed9aae4a0b2d2c80d)

File: client/my-sites/earn/memberships/index.jsx
Comment (romarioraffington) [reply client/my-sites/earn/memberships/index.jsx:514]: > I haven't tested, but that seems like it should break the original flow by not diaplaying the notice....

It shouldn't. Once we pass `calypso` as a query param for `source`,  `earn` will be returned as a query param for the redirect. See D103932-code for more context.

Comment (romarioraffington) [reply client/my-sites/earn/memberships/index.jsx:514]: > Can we introduce constants for that as well?

We could! I didn't add it as a const because it's only being used in one file.

**Comment by romarioraffington** [comment]
Note to self: Do not deploy until https://github.com/Automattic/jetpack/pull/29351 is released to WPCOM

File: client/my-sites/earn/memberships/index.jsx
Comment (millerf) [reply client/my-sites/earn/memberships/index.jsx:514]: > We could! I didn't add it as a const because it's only being used in one file.

Oh, ok...

Comment (millerf) [reply client/my-sites/earn/memberships/index.jsx:514]: > It shouldn't. Once we pass calypso as a query param for source, earn will be returned as a query param for the redirect. See D103932-code for more context.

Ok, it kind of makes sense now. But this is not very straightforward....

**Review by millerf — APPROVED** [review]
LGTM as long as the tests pass.... I'll relaunch them to see it is was a fluke...

File: client/state/memberships/settings/actions.js
Comment (n3f) [inline client/state/memberships/settings/actions.js:8]: Should this be defaulted to `calypso` here? (So any current uses keep working).

Ninja edit: I couldn't find any other places where it was used so 🤷‍♂️

**Pull request merged**
